### DISCUSSION TITLE
Dirty Homebrew version, HOMEBREW_VERSION 3.2.10-6-g3947d09-dirty

### DISCUSSION THREAD

**Discussion by LGitHub-sprout** [Homebrew/discussions#2067] [Getting started] [answered]
Hi,

I've searched Homebrew/discussions and read a couple posts w similar -dirty version, but didn't see specific solution or info if this should be fixed or can be left alone.

I had older version of both Homebrew and git on my MBPro 2015 Big Sur (11.5.1) and ran the following before running `brew update`

`$ brew --version`
was not up-to-date ( < 3.0.0 ). I don’t recall whether that version was also “-dirty”. 
I researched how to update brew and found this:
https://osxdaily.com/2021/02/13/how-update-homebrew-mac/
I then confirmed here:
https://docs.brew.sh/FAQ#how-do-i-update-my-local-packages
so far, so good. 
I then ran `git --version`, `brew outdated`, `brew list` commands and all needed updating.

I then ran `brew update` and now noticed "-dirty" version. I'd like to know if this needs fixing or is OK to move fwd.  

Additionally, I got a warning about uncommitted Homebrew files which I researched [here](https://github.com/Homebrew/discussions/discussions/675) and then ran
`cd /usr/local/Homebrew && git clean -d -f -x` (stash did nothing) which cleared all files except 2 as follows:

M docs/Manpage.md
UU manpages/brew.1

Should I run suggested commands [here](https://docs.brew.sh/Common-Issues#brew-update-complains-about-untracked-working-tree-files) on these or are they OK to leave and move fwd?

Many thanks for any help. I've spent a lot of time researching and reading which has increased my confidence a bit but would like to resolve this ASAP now. Thank you

**Comment by SMillerDev** [comment, marked as answer]
> Should I run suggested commands here on these or are they OK to leave and move fwd?

Seeing how you're having the issue described in the FAQ entry, running the commands provided as the solution sounds like the best course of action.

Regarding your main issue of the `-dirty` suffix, that does not mean anything on it's own. You can just ignore it unless you have problems with brew.

**Reply by LGitHub-sprout** [reply]
I appreciate the quick reply, THANK YOU.
I ran 
`cd $(brew --repository)
git reset --hard FETCH_HEAD` per above mentioned "Common Issues" link.
which cleared the warning and gave me a "Your system is ready to brew." message from brew doctor. 
Bonus... my HOMEBREW_VERSION no longer says '-dirty'
Well done... and much appreciated. 
Have a great day.
